libraries/dotnet/EnergyOriginEventStore/EnergyOriginEventStore/EventStore/Memory/MemoryEventStore.cs

```diff
@@ -0,0 +1,133 @@
+using System.Collections.Concurrent;
+using EnergyOriginEventStore.EventStore.Serialization;
+using Newtonsoft.Json;
+
+namespace EnergyOriginEventStore.EventStore.Memory;
+
+public class MemoryEventStore : IEventStore
+{
+    private List<MessageEventArgs> _messages = new();
+    private ConcurrentQueue<Action> _actions = new();
+    [ThreadStatic] private bool isDraining = false;
```

Comment:
See https://www.jetbrains.com/help/resharper/ThreadStaticAtInstanceField.html for more info

```suggestion
    [ThreadStatic] private static bool isDraining = false;
```

Reply:
Since it is used to lock a critical section with async/await keywords, then this should be a `new SemaphoreSlim(1,1)`